include/cpp2util.h

```diff
@@ -87,7 +87,11 @@
         #include <variant>
         #include <version>
         #include <memory>
-        #include <memory_resource>
+        #if __has_include(<memory_resource>)
```

Comment:
need to check `#if defined __has_include`

Reply:
I was thinking if should I add that check before using it. If `__has_include` is not present it means that we are dealing with so old C++ that it will not compile the rest of the file.

But you are right. For the reason of completeness I will add that.

Reply:
I'd wait for the owner's opinion before doing that.

Reply:
Actually, I forgot that `__has_include` is standard as of C++17, so I retract my comment.

Reply:
It might still be a good idea to `#error` if the fallback fails, too. Better to fail when you know there's a problem than later, on a missing symbol error or some such.